src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,202 @@
+package duke;
+
+import duke.command.AddTaskCommand;
+import duke.command.Command;
+import duke.command.DeleteCommand;
+import duke.command.ExitCommand;
+import duke.command.FindCommand;
+import duke.command.MarkDoneCommand;
+
+import duke.task.DeadlineTask;
+import duke.task.EventTask;
+import duke.task.TaskList;
+import duke.task.TodoTask;
+
+import java.time.DateTimeException;
+
+public class Parser {
+
+    private Duke duke;
+    private TaskList taskList;
+    private Storage storage;
+    private Ui ui;
+
+    /**
+     * Constructor for a Parser for Duke.
+     * @param duke Duke instance the Parser is being used for.
+     * @param taskList TaskList used by the Duke.
+     * @param storage Storage used by the Duke.
+     * @param ui Ui used by the Duke.
+     */
+    public Parser(Duke duke, TaskList taskList, Storage storage, Ui ui) {
+        this.duke = duke;
+        this.taskList = taskList;
+        this.storage = storage;
+        this.ui = ui;
+    }
+
+    /**
+     * Parses the user input string.
+     * @param input user input
+     * @return true if user enters the exit command, false otherwise.
+     */
+    public Command parse(String input) {
+        String[] splitInput = input.split(" ", 2);
+        String firstWord = splitInput[0];
+
+        switch (firstWord) {
+        case "bye":
+            return new ExitCommand(taskList, storage, ui, duke);
+        case "delete":
+            return deleteFromList(splitInput);
+        case "deadline":
+            return addDeadline(splitInput);
+        case "done":
+            return setTaskDone(splitInput);
+        case "list":
+            ui.showMessage(taskList.toString());
+            break;
+        case "event":
+            return addEvent(splitInput);
+        case "find":
+            return findTask(splitInput);
+        case "todo":
+            return addTodo(splitInput);
+        default:
+            ui.showError("Error: Command not recognised, sorry.");
+            break;
+        }
+        return null;
+    }
+
+    /**
+     * Parses a user input to add a Todo type task.
+     * @param input user input split into individual words.
+     * @return Command to add a new TodoTask, null if error occurs.
+     */
+    private Command addTodo(String[] input) {
+        try {
+            String name = input[1];
```

Comment:
Perhaps it would be better to name this variable `description` as it is more descriptive of what it is representing.